core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -212,6 +225,70 @@ class half_t {
   KOKKOS_FUNCTION
   half_t(unsigned long long rhs) : val(cast_to_half(rhs).val) {}
 
+  // Atomics
+  KOKKOS_FUNCTION
+  void atomic_add(const volatile half_t src) const volatile {
```

Comment:
either copy the content to where you use it (it looks like its only called once?) or rename to something like impl_volatile_add.